renderer/renderer.go

```diff
@@ -289,11 +302,46 @@ func RenderGemini(s string, width int, proxied bool) (string, []string) {
 	pre := false
 	buf := "" // Block of regular or preformatted lines
 
+	// Language, formatter, and style for syntax highlighting
+	lang := ""
+	formatterName := TermColor
+	styleName := viper.GetString("a-general.highlight_style")
+
 	// processPre is for rendering preformatted blocks
 	processPre := func() {
 
+		// Perform syntax highlighting if language is set
+		if lang != "" {
+			style := styles.Get(styleName)
+			if style == nil {
+				style = styles.Fallback
+			}
+			formatter := formatters.Get(formatterName)
+			if formatter == nil {
+				formatter = formatters.Fallback
+			}
+			lexer := lexers.Get(lang)
+			if lexer == nil {
+				lexer = lexers.Fallback
+			}
+
+			// Strip existing ANSI codes
+			strippedBuf := ansiRegex.ReplaceAllString(buf, "")
```

Comment:
Can't `buf` be reused for this, reducing memory usage?

Reply:
The ANSI codes are only stripped if there is no error formatting the text using the detected language; if there is an error we just display any existing ANSI codes and treat it as normal preformatted text. That's why I was keeping buf around.

To possibly reduce memory usage I can remove 329 and change 332 to the following:
```
iterator, err := lexer.Tokenise(nil, ansiRegex.ReplaceAllString(buf, ""))
```

I thought that maybe the go compiler would optimize out strippedBuf, but after a quick search it sounds like some people complain that go sometimes doesn't do enough optimization. This should be a simple way to ensure that we avoid creating extra duplicates of buf.